service/history/timerGate.go

```diff
@@ -35,14 +35,19 @@ type (
 		// Update update the timer gate, return true if update is a success
 		// success means timer is idle or timer is set with a sooner time to fire
 		Update(nextTime time.Time) bool
+	}
+
+	// LocalTimerGate interface
+	LocalTimerGate interface {
```

Comment:
Any reason Close is not needed by remote timer gate?  Both local and remote timers should have the same life-cycle.

Reply:
local is implemented using a go timer, which can be closed.

remote is purely driven by events, and there is no background coroutine  involved.